metricbeat/module/windows/perfmon/pdh_windows.go

```diff
@@ -368,38 +372,37 @@ func (r *PerfmonReader) Read() ([]common.MapStr, error) {
 	}
 
 	// Write the values into the map.
-	result := make([]common.MapStr, 0, len(values))
-	var errs multierror.Errors
-
-	for counterPath, counter := range values {
-		for _, val := range counter {
-			ev := common.MapStr{}
-			instanceKey := r.instanceLabel[counterPath]
-			ev.Put(instanceKey, val.Instance)
-			measurementKey := r.measurement[counterPath]
-			ev.Put(measurementKey, val.Measurement)
-
-			if val.Err != nil {
-				switch val.Err {
-				case PDH_CALC_NEGATIVE_DENOMINATOR:
-				case PDH_INVALID_DATA:
-					if r.executed {
-						errs = append(errs, errors.Wrapf(val.Err, "key=%v", measurementKey))
-					}
-				default:
-					errs = append(errs, errors.Wrapf(val.Err, "key=%v", measurementKey))
-				}
+	events := make([]mb.Event, 0, len(values))
+
+	for counterPath, values := range values {
+		for _, val := range values {
+			if val.Err != nil && !r.executed {
+				r.log.Debugw("Ignoring the first measurement because the data isn't ready",
+					"error", val.Err, logp.Namespace("perfmon"), "query", counterPath)
+				continue
 			}
 
-			result = append(result, ev)
-		}
-	}
+			event := mb.Event{
+				MetricSetFields: common.MapStr{},
+				Error:           errors.Wrapf(val.Err, "failed on query=%v", counterPath),
+			}
+
+			if val.Instance != "" {
+				event.MetricSetFields.Put(r.instanceLabel[counterPath], val.Instance)
```

Comment:
@exekias I wonder if this would also be a place to use safteput. Also below.

Reply:
I don't have much knowledge about Windows perfom metrics, in general `safemapstr.Put` would be required when we know the field may contain dots and these don't necessarily imply nesting. Basically when field names can cause a nesting issue like this:

```
field1.total: 3
field1: "foo"
```

Reply:
I don't believe `safemapstr.Put` is very applicable here (I did consider it during this fix). The user has full control over the field names plus there is only one metric in an event.